js/components/lab-model.js

```diff
@@ -20,9 +20,7 @@ export default class LabModel extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.model !== prevProps.model) {
-      this.setState({loading: true});
-    }
+    this.setState({ loading: (this.props.model !== prevProps.model) });
```

Comment:
If you wanted to restore previous behaviour using more concise form, note that this is quite different logic.
We want to set `loading: true` here, but we don't want to set `loading: false`. The fact that props.model hasn't changed doesn't mean that model actually has loaded. Also, if you try to follow what happens here, you'll see that this line will never actually have any effect - first, you set `loading` to true. That triggers componentDidUpdate again, but this time `model` prop isn't updated, so we immediately set `loading` back to false.

But more important comment: you shouldn't call `setState` in `componentDidMount` like this, without checking some condition first. That might lead to infinite loop of component updates. Note that setState leads to componentDidUpdate. In this case it doesn't happen because this component extends PureComponent (https://facebook.github.io/react/docs/react-component.html#shouldcomponentupdate) instead of regular Component.